config/routes.rb

```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'products/index'
+  resources :products , only: [:index]
```

Comment:
Now, do you really need to define this route? The `root_path` points to this controller and action, right?

Reply:
Ah yes! We cleaned up the code to only have one path.

Reply:
Yes, but you did not answer my question. Do you really need to define this route? The root_path points to this controller and action, right?

Reply:
Ok. So when we comment out the `root controller: :products, action: :index`, but keeping the `get 'products/index'` we get an error message about an undefined local variable or method for `root_path`. But when we do the other way around, when we comment out the `get 'products/index'` but keeping the `root controller: :products, action: :index` activated, our tests go green and the root path is found. Meaning that the line for `get 'products/index'` does not seem to be needed. We don't know exactly why, but we suspect that this has to do with that the we have defined Product as resources already, and then Product need to have a root path defined for controller and action specifically?!

Reply:
What IF you just keep the root setting and comment out the other two routes?

Reply:
Then the test passes! Ok, so I don't think it has to do with resource "definition" on line 3.

When we check the defined `rails routes` with both only having `get` - and then only the `root setting` non-commented respectively - we get NO "root" direction without the `root setting` path defined. You have a path for products_index, but none for root... SO: Unless we define the root path using the `root controller` syntax, we cannot use it in our test, since the `root_path` is pointing to that.

Reply:
So what is the conclusion?

Reply:
Since this `root controller: :products, action: :index` does the job, we take the rest away. (At least for now, not sure what happens when we need to make other actions available but I guess we cross that bridge when we get there)

Reply:
Thank you, that was what I was looking for.